pkg/diffmanager/setup_test.go

```diff
@@ -0,0 +1,235 @@
+			}
+			return
+		}
+		if location == "/notfound/" {
+			expErr := errors.New("non 200 response")
+			if err.Error() != expErr.Error() {
+				t.Errorf("Expected error: %s, received error: %s", expErr, err)
+			}
+			return
+		}
+		expResp := testutils.GetGitDiff()
+		if err != nil {
+			t.Errorf("error in getting changed files, error %v", err.Error())
+		} else if gitprovider == "github" && !reflect.DeepEqual(resp, expResp) {
+			t.Errorf("Expected: %+v, received: %+v", expResp, resp)
+		} else if gitprovider == "gitlab" && len(resp) != 17 {
+			t.Errorf("Expected map entries: 17, received: %v, received map: %v", len(resp), resp)
+		}
+	}
+
+	t.Run("TestDiffManager: PRdiff for github gitprovider", func(t *testing.T) {
+		checkPRdiff(t, "/testdata", "github")
+	})
+	t.Run("TestDiffManager: PRdiff for gitlab gitprovider", func(t *testing.T) {
+		checkPRdiff(t, "/testdata", "gitlab")
+	})
+	t.Run("TestDiffManager: PRdiff for unsupported gitprovider", func(t *testing.T) {
+		checkPRdiff(t, "/testdata", "")
+	})
+	t.Run("TestDiffManager: PRdiff for non 200 status", func(t *testing.T) {
+		checkPRdiff(t, "/notfound/", "github")
+	})
+
+}
+
+func Test_GetChanagedFilesForGithubCommitDiff(t *testing.T) {
+	server := httptest.NewServer( // mock server
+		http.FileServer(http.Dir("../../testutils")), // mock data stored at testutils/testdata/index.txt
+	)
+	defer server.Close()
+
+	logger, err := testutils.GetLogger()
+	if err != nil {
+		t.Errorf("Can't get logger, received: %s", err)
+	}
+	config, err := testutils.GetConfig()
+	if err != nil {
+		t.Errorf("Can't get logger, received: %s", err)
+	}
+	dm := NewDiffManager(config, logger)
+
+	checkCommitDiff := func(t *testing.T, location, baseCommit, targetCommit, gitProvider string) {
+		t.Helper()
+		p, err := testutils.GetPayload()
+		if err != nil {
+			t.Errorf("Unable to get payload, error %v", err)
+		}
+		p.RepoLink = server.URL
+		p.EventType = "push"
+		p.PullRequestNumber = 2
+		p.GitProvider = gitProvider
+		p.RepoLink = server.URL + location
+		cloneToken := ""
+		p.BaseCommit = baseCommit
+		p.TargetCommit = targetCommit
+		global.APIHostURLMap[p.GitProvider] = server.URL
+		resp, err := dm.GetChangedFiles(context.TODO(), p, cloneToken)
+		if gitProvider == "gittest" {
+			if resp != nil || err == nil {
+				t.Errorf("Expected error: 'unsupoorted git provider', received: %v\nexpected response: nil, received: %v", err, resp)
+			}
+			return
+		}
+		if baseCommit == "" || location == "/notfound/" {
+			if err != nil || resp != nil {
+				t.Errorf("Received error: %v, response: %v", err, resp)
+			}
+			return
+		}
+		expResp := make(map[string]int)
+		if err != nil {
+			t.Errorf("error in getting changed files, error %v", err.Error())
+		} else if !reflect.DeepEqual(resp, expResp) {
+			t.Errorf("Expected: %+v, received: %+v", expResp, resp)
+		}
+	}
+	t.Run("TestDiffManager: Commitdiff", func(t *testing.T) {
```

Comment:
I think we can use table driven tests instead of running, multiple sub tests like this. Will be much cleaner

Reply:
Agree it will be much cleaner by using table driven tests, but for each subtest in table driven test, I will have to provide all the fields of different structs like Payload which are being passed to test that particular function.